utils/kubernetes/kubeutil.py

```diff
@@ -230,6 +243,20 @@ def retrieve_metrics(self):
         """
         return retrieve_json(self.metrics_url)
 
+    def get_deployment_for_replicaset(self, rs_name):
+        """
+        Get the deployment name for a given replicaset name
+        For now, the rs name's first part always is the deployment's name, see
+        https://github.com/kubernetes/kubernetes/blob/release-1.6/pkg/controller/deployment/sync.go#L299
+        But it might change in a future k8s version. The other way to match RS and deployments is comparing
+        label selectors, which would be very costly, especially matchExpressions matching.
+        """
+        end = rs_name.rfind("-")
```

Comment:
As the code uses rfind, it will match on the last dash or the name. The second part (spechash) can't include a dash.